lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -143,6 +145,21 @@ pub trait SegmentOptimizer {
             .copied()
             .unwrap_or(0);
 
+        let mut available_space;
+        match fs4::available_space(self.temp_path()) {
+            Some(size) => available_space = size,
+            Err(error) => CollectionError::service_error(format!(
+                "Failed to fetch available space for path: {} due to: {}",
+                self.temp_path().to_owned(),
+                error.to_string()
+            )),
```

Comment:
Some file systems might not support it, so we would prefer to go on with the optimization instead of failing. 
So logging the error should be enough

Reply:
Sure I will update the changes.